db/hints/resource_manager.cc

```diff
@@ -52,7 +52,8 @@ future<bool> is_mountpoint(const fs::path& path) {
 
 future<semaphore_units<named_semaphore::exception_factory>> resource_manager::get_send_units_for(size_t buf_size) {
     // Let's approximate the memory size the mutation is going to consume by the size of its serialized form
-    size_t hint_memory_budget = std::max(_min_send_hint_budget, buf_size);
+    const size_t max_send_hint_budget = _max_send_in_flight_memory / std::max(uint32_t(1), _max_hints_send_queue_length());
```

Comment:
This is a bit confusing. Previously the variable was called `_min_send_hint_budget` and was used with `std::max` which means it was a lower bound for the budget. Now the variable is called `max_send_hint_budget` which creates an impression that it's an upper bound but it's still used in `std::max` so it's a lower bound.

Reply:
My mistake. The variable should be named `min_send_hint_budget`. Apart from dropping the underscore, I accidentally changed min to max - not sure why. I'll fix it after the dtests finish.

Reply:
Done, changed to `min_send_hint_budget`. As a bonus, I added a comment which explains why the hint send budget is calculated this way.